app/Http/Controllers/PhotoController.php

```diff
@@ -122,23 +126,25 @@ public function setTitle(PhotoIDsRequest $request, SetTitle $setTitle)
 	/**
 	 * Set if a photo is a favorite.
 	 *
-	 * @param Request $request
+	 * @param PhotoIDsRequest $request
+	 * @param SetStar         $setStar
 	 *
 	 * @return string
 	 */
-	public function setStar(PhotoIDsRequest $request, SetStar $setStar)
+	public function setStar(PhotoIDsRequest $request, SetStar $setStar): string
 	{
-		return $setStar->do(explode(',', $request['photoIDs']), $request['title']) ? 'true' : 'false';
+		return $setStar->do(explode(',', $request['photoIDs'])) ? 'true' : 'false';
```

Comment:
don't you want to refactor that one similarly to the albums:
```php
		$setStar->do(explode(',', $request['photoIDs']);

		return response()->noContent();
```

Reply:
My guess is that @nagmat84 started with the refactoring of the responses but once he realized the true scale of the problem :smiley: he postponed it until a future PR.

Reply:
@kami4 is totally right. I have another PR in the pipeline and I am working on another (local) branch `refactor_return` which deals with consistent return types as proper JSON in all cases, cleaning up `JsonError` and `JsonWarning`, etc. So stay tuned ;-)

This was only an intermediate fix to keep the code working without putting too much in a single PR.

(OK, the latter statement sounds ridiculous given the extent of this PR.)